docs/user-manual/core-tools.md

```diff
@@ -219,12 +219,354 @@ As c++ is a strongly typed language, the signal template parameters signature ha
 
 Note also that the *dqm4hep::core::Signal* class allows to :
 
-- connect methods from any c++ class.
-- connect methods with unlimited number of arguments. Nevertheless, the parameter types have to be specified in the template arguments of the *dqm4hep::core::Signal* object : 
+- connect methods from any c++ class,
+- connect global functions,
+- disconnect these functions if needed,
+- connect methods or functions with unlimited number of arguments. Nevertheless, the parameter types have to be specified in the template arguments of the *dqm4hep::core::Signal* object : 
 
 ```cpp
   Signal<std::string, int, int, double, float, int> aLongSignal;
 ``` 
 
+# Status code and useful pre-processor macros
+
+If you navigate through the DQM4hep code, you will notice the use of the *StatusCode* enum, the *StatusCodeException* and pre-processor macros such as *RETURN_RESULT_IF()*.
+The *StatusCode* enum contains the folowing values:
+
+```cpp
+enum StatusCode {
+  STATUS_CODE_SUCCESS,
+  STATUS_CODE_FAILURE,
+  STATUS_CODE_NOT_FOUND,
+  STATUS_CODE_NOT_INITIALIZED,
+  STATUS_CODE_ALREADY_INITIALIZED,
+  STATUS_CODE_ALREADY_PRESENT,
+  STATUS_CODE_OUT_OF_RANGE,
+  STATUS_CODE_NOT_ALLOWED,
+  STATUS_CODE_INVALID_PARAMETER,
+  STATUS_CODE_UNCHANGED,
+  STATUS_CODE_INVALID_PTR
+};
+```
+
+The *StatusCodeException* exception class handle one of these status codes on creation. Together with the StatusCode enum are also defined the following pre-processor macros:
+
+- *RETURN_RESULT_IF(code, Operator, Command)*: return the status code returned by the command if the comparison code with *code* using the operator *Operator* fails. See example below.
```

Comment:
The phrasing for both `RETURN_RESULT_IF` and `RETURN_RESULT_IF_AND_IF` are a little hard to follow. Maybe something more like:

> Runs the _command_, and returns it's status code if the comparison with _code_ using the _operator_ is false.
> Runs the _command_, and returns it's status code if the comparison with either _code1_ or _code2_ using the _operator_ is false.

Reply:
it's -> its, mister ! ^^